index.js

```diff
@@ -23,6 +23,80 @@ server.get("/api/projects", (req, res) => {
     .catch(() => res.status(500).json({ message: "Server Error" }));
 });
 
+server.post("/api/projects", (req, res) => {
+  const { name, description } = req.body;
+
+  if (!name || !description) {
+    res.status(422).json({ message: "Name and Description required!" });
+  } else {
+    db("projects")
+      .insert({ name, description })
+      .then(result => {
+        if (result.rowCount) {
+          db("projects").then(projects => {
+            res.json(projects);
+          });
+        } else {
+          res.status(400).json({ message: "Failed to add project" });
+        }
+      })
+      .catch(() => {
+        res.status(500).json({ message: "Server Error" });
+      });
+  }
+});
+
+server.post("/api/actions", (req, res) => {
+  const { description, project_id, notes } = req.body;
+
+  if (!description || !project_id) {
+    res.status(422).json({ message: "Description and Project ID Required!" });
+  } else {
+    db("projects")
+      .where("id", project_id)
+      .first()
+      .then(project => {
+        if (project) {
+          db("actions")
+            .insert({ description, project_id, notes })
+            .then(result => {
+              if (result.rowCount) {
+                db("actions").then(actions => {
+                  res.json(actions);
+                });
+              } else {
+                res.status(400).json({ message: "Failed to add action" });
+              }
+            });
+        } else {
+          res.status(400).json({ message: "Project by that id was not found" });
+        }
+      })
+      .catch(() => res.status(500).json({ message: "Server Error" }));
```

Comment:
When dealing with `catch` your code is usually more easily troubleshooted when you include the supply the error code and message from params `.catch(({ code, message }) => res.status(code).json({ message })`